terraso_backend/apps/soil_id/graphql/soil_id/resolvers.py

```diff
@@ -0,0 +1,272 @@
+        soil_info=resolve_soil_info(soil_match),
+    )
+
+
+def resolve_location_matches_from_soil_id_result(soil_list_json: dict):
+    matches = []
+    for match in soil_list_json["soilList"]:
+        if match["id"]["rank_loc"] != "Not Displayed":
+            matches.append(resolve_location_based_soil_match(match))
+    return LocationBasedSoilMatches(matches=matches)
+
+
+def resolve_location_based_soil_matches(_parent, _info, latitude: float, longitude: float):
+    # TODO: remove this line to re-enable using the actual algorithm to resolve this query
+    return dummy_location_matches
+
+    result = list_soils(lat=latitude, lon=longitude)
+
+    if isinstance(result, str):
+        return None
+
+    return resolve_location_matches_from_soil_id_result(result.soil_list_json)
+
+
+def resolve_data_based_soil_match(soil_matches: list[dict], ranked_match: dict):
+    soil_match = [
+        match
+        for match in soil_matches
+        if int(match["site"]["siteData"]["componentID"]) == ranked_match["componentID"]
+    ][0]
+    site_data = soil_match["site"]["siteData"]
+
+    return DataBasedSoilMatch(
+        data_source=site_data["dataSource"],
+        distance_to_nearest_map_unit_m=site_data["minCompDistance"],
+        location_match=resolve_soil_match_info(ranked_match["score_loc"], ranked_match["rank_loc"]),
+        data_match=resolve_soil_match_info(ranked_match["score_data"], ranked_match["rank_data"]),
+        combined_match=resolve_soil_match_info(
+            ranked_match["score_data_loc"], ranked_match["rank_data_loc"]
+        ),
+        soil_info=resolve_soil_info(soil_match),
+    )
+
+
+def parse_texture(texture: Optional[DepthDependentSoilData.Texture]):
+    if texture is None:
+        return None
+    return texture.value.replace("_", " ").lower()
+
+
+def parse_rock_fragment_volume(
+    rock_fragment_volume: Optional[DepthDependentSoilData.RockFragmentVolume],
+):
+    if rock_fragment_volume is None:
+        return None
+    elif rock_fragment_volume == DepthDependentSoilData.RockFragmentVolume.VOLUME_0_1:
+        return "0-1%"
+    elif rock_fragment_volume == DepthDependentSoilData.RockFragmentVolume.VOLUME_1_15:
+        return "1-15%"
+    elif rock_fragment_volume == DepthDependentSoilData.RockFragmentVolume.VOLUME_15_35:
+        return "15-35%"
+    elif rock_fragment_volume == DepthDependentSoilData.RockFragmentVolume.VOLUME_35_60:
+        return "35-60%"
+    else:
+        return ">60%"
+
+
+def parse_color_LAB(color_LAB: Optional[LABColorInput]):
+    if color_LAB is None:
+        return None
+    return [color_LAB.L, color_LAB.A, color_LAB.B]
+
+
+def parse_surface_cracks(surface_cracks: SoilData.SurfaceCracks):
+    if surface_cracks is None:
+        return None
+    return surface_cracks == SoilData.SurfaceCracks.DEEP_VERTICAL_CRACKING
+
+
+def parse_rank_soils_input_data(data: SoilIdInputData):
+    # TODO: pass in values for elevation and bedrock
```

Comment:
question: is this work tracked?

Reply:
no, i don't think so, but this isn't new code